File: output/CodeGen/codenet/Java/Python/s319823779.py
def main():
	sc = Scanner(System.in)
	a = sc.nextInt()
	b = sc.nextInt()
	c = sc.nextInt()
	x = sc.nextInt()
	
	res = 0
	
	for i in range(0, a+1):
		for j in range(0, b+1):
			for k in range(0, c+1):
				total = 500*i + 100*j + 50*k
				
				if total == x:
					res += 1
	
	print(res)

if __name__ == '__main__':
	main()

File: output/ChatGPT/codenet/Java/C++/s319823779.cpp
#include <iostream>
using namespace std;

int main() {
    int a, b, c, x;
    cin >> a >> b >> c >> x;
    
    int res = 0;
    
    for(int i=0 ; i<=a ; i++){
        for(int j=0 ; j<=b ; j++){
            for(int k=0 ; k<=c ; k++){
                int total = 500*i + 100*j + 50*k;
                
                if( total==x ){
                    res++;
                }
            }
        }
    }
    
    cout << res << endl;
    
    
    return 0;
}